src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsRequestBodyFilter.cs

```diff
@@ -13,97 +13,165 @@ public class XmlCommentsRequestBodyFilter : IRequestBodyFilter
+        {
+            var (summary, example) = GetPropertyTags(propertyInfo);
+
+            if (summary is not null)
+            {
+                requestBody.Description = summary;
             }
 
-            var example = exampleNode.ToString();
+            if (requestBody.Content?.Count is 0)
+            {
+                return;
+            }
 
             foreach (var mediaType in requestBody.Content.Values)
             {
                 mediaType.Example = XmlCommentsExampleHelper.Create(context.SchemaRepository, mediaType.Schema, example);
             }
         }
 
-        private void ApplyParamTags(OpenApiRequestBody requestBody, RequestBodyFilterContext context, ParameterInfo parameterInfo)
+        private (string summary, string example) GetParamTags(ParameterInfo parameterInfo)
         {
             if (parameterInfo.Member is not MethodInfo methodInfo)
             {
-                return;
+                return (null, null);
             }
-
-            // If method is from a constructed generic type, look for comments from the generic type method
             var targetMethod = methodInfo.DeclaringType.IsConstructedGenericType
                 ? methodInfo.GetUnderlyingGenericTypeMethod()
                 : methodInfo;
 
             if (targetMethod is null)
             {
-                return;
+                return (null, null);
             }
-
             var methodMemberName = XmlCommentsNodeNameHelper.GetMemberNameForMethod(targetMethod);
 
-            if (!_xmlDocMembers.TryGetValue(methodMemberName, out var propertyNode)) return;
-
+            if (!_xmlDocMembers.TryGetValue(methodMemberName, out var propertyNode))
+            {
+                return (null, null);
+            }
             var paramNode = propertyNode.SelectFirstChildWithAttribute("param", "name", parameterInfo.Name);
 
-            if (paramNode is not null)
+            if (paramNode is null)
+            {
+                return (null, null);
+            }
+
+            var summary = XmlCommentsTextHelper.Humanize(paramNode.InnerXml);
+            var example = paramNode.GetAttribute("example");
+
+            return (summary, example);
+
+        }
+
+        private void ApplyParamTagsForBody(OpenApiRequestBody requestBody, RequestBodyFilterContext context, ParameterInfo parameterInfo)
+        {
+            var (summary, example) = GetParamTags(parameterInfo);
+
+            if (summary is not null)
             {
-                requestBody.Description = XmlCommentsTextHelper.Humanize(paramNode.InnerXml);
+                requestBody.Description = summary;
+            }
+
+            if (requestBody.Content?.Count is 0)
+            {
+                return;
+            }
 
-                var example = paramNode.GetAttribute("example");
-                if (!string.IsNullOrEmpty(example))
+            if (!string.IsNullOrEmpty(example))
```

Comment:
Could make this part of the previous early return as an `||` and them avoid this extra level of indentation?